.github/workflows/rust.yml

```diff
@@ -70,6 +70,18 @@ jobs:
     - uses: dtolnay/rust-toolchain@1.60.0
     - run: cargo build
 
+  build-nostd:
+    name: Build on no_std target (thumbv7em-none-eabi)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          targets: thumbv7em-none-eabi
+      - run: cargo build --target thumbv7em-none-eabi --release --no-default-features --features alloc
```

Comment:
Maybe add a `--no-default-features` without `--features`?

Reply:
Sure good idea - though I think we may want two where 

1) The first is all features with no_std that doesn't require alloc and 
2) The second that has features requires alloc ?

So it tests both no_std without alloc and with alloc ?

Though there isn't target available for no_alloc or mechanism to make alloc unavailable so we may just put them all in one test ?

Reply:
I still think it's good to test that it builds without any features, which is at least an additional third test

Reply:
k. I will leave serde exclude in the comment and if this is rebased in the fix it can be removed to check serde as well so we don't break the main

Reply:
Added so `cargo-hack` is the second where as vanilla `cargo build` `--no-default-features` is the first.